plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -138,39 +140,70 @@ compileLiteral = \case
 -- do different things to the inner expression. This one assumes it's a literal, the other one keeps compiling
 -- through it.
 
-data StringExprContentAs = AsBytes | AsText
+data EncodingAction = UndoUtf8 | PreserveUtf8
 
 -- | Get the bytestring content of a string expression, if possible.
 -- Follows (Haskell) variable references!
-stringExprContent :: StringExprContentAs -> GHC.CoreExpr -> Maybe BS.ByteString
-stringExprContent contentAs coreExpr = case coreExpr of
+stringLiteralBytes :: EncodingAction -> GHC.CoreExpr -> Maybe BS.ByteString
+stringLiteralBytes encodingAction coreExpr = case coreExpr of
   GHC.Lit (GHC.LitString bytes) ->
     Just bytes
   GHC.Var isUnpackCString `GHC.App` GHC.Lit (GHC.LitString bytes)
     | GHC.getName isUnpackCString == GHC.unpackCStringName ->
         Just bytes
   GHC.Var isUnpackCStringUtf8 `GHC.App` GHC.Lit (GHC.LitString bytes)
     | GHC.getName isUnpackCStringUtf8 == GHC.unpackCStringUtf8Name ->
-    case contentAs of
-      AsText -> Just bytes
-      AsBytes ->
+    case encodingAction of
+      PreserveUtf8 -> Just bytes
+      UndoUtf8 ->
         -- GHC stores bytestring literals UTF-8 encoded, decoding them at runtime.
         -- In Plinth we decode such bytestrings in compile-time.
         BS.pack <$> fromUtf8 (BS.unpack bytes)
   -- See Note [unpackFoldrCString#]
   GHC.Var build `GHC.App` _ `GHC.App` GHC.Lam _ (GHC.Var unpack `GHC.App` _ `GHC.App` expr)
     | GHC.getName build == GHC.buildName && GHC.getName unpack == GHC.unpackCStringFoldrName ->
-      stringExprContent contentAs expr
+      stringLiteralBytes encodingAction expr
   -- GHC helpfully generates an empty list for the empty string literal instead of a 'LitString'
   GHC.Var nil `GHC.App` GHC.Type (GHC.tyConAppTyCon_maybe -> Just tc)
     | nil == GHC.dataConWorkId GHC.nilDataCon, GHC.getName tc == GHC.charTyConName ->
       Just mempty
   -- Chase variable references! GHC likes to lift string constants to variables,
   -- that is not good for us!
   GHC.Var (GHC.maybeUnfoldingTemplate . GHC.realIdUnfolding -> Just unfolding) ->
-    stringExprContent contentAs unfolding
+    stringLiteralBytes encodingAction unfolding
   _ -> Nothing
 
+-- | Given a GHC Core expression representing a string literal
+-- extracts a ByteString from it.
+stringLiteralAsBytes
```

Comment:
Please think of better names for `stringLiteralBytes` and `stringLiteralAsBytes`. Their main difference is that one returns `Maybe` and the other throws an error. The presence or absence of "as" does not convey such information whatsoever.

Reply:
Ok, you want to highlight this difference in behaviour by reflecting it in the name. 
How do you like 
* `tryStringLiteralBytes`  <- maybe-returning 
* `stringLiteralBytes` <- throwing
?